### ISSUE TITLE
Document how to add request data to the SLF4J MDC

### ISSUE THREAD

**Issue by ianfp** [micronaut-projects/micronaut-core#7914]
### Issue description

I'm trying to add per-request fields to the SLF4J MDC. For example:
```
{
    "requestName": "GET /chickens",
    "traceId": "some-trace-id",
    // etc
}
```

I've tried using an `HttpServerFilter`, but those run on the event loop thread, whereas my requests are handled by the I/O thread pool per this configuration:
```yaml
micronaut:
  server:
    thread-selection: io
```
As a result, the Thread-local MDC is not propagated across to the request-handling thread. My application is not using reactive/flux stuff.

I cannot find any documentation about how to do this. I found this page:
https://micronaut-projects.github.io/micronaut-tracing/4.2.1/guide/index.html
but I can't see how it answers my question.

I found some seemingly-related issues, but none of them led me to a solution:
- https://github.com/micronaut-projects/micronaut-core/issues/3106
- https://github.com/micronaut-projects/micronaut-core/issues/2312
- https://github.com/micronaut-projects/micronaut-core/issues/1669

There's even a StackOverflow question about this, but it does not have any answers:
- https://stackoverflow.com/questions/68217927/add-a-request-id-to-slf4j-logging-in-micronaut
The one comment is not sufficiently detailed to be helpful.

**Comment by PavelGolubTR** [comment]
I see that this issue is quite old one and I assume that someone has found solution for the problem. Would be really helpful to get direction and details on it.

**Comment by francesco-monorchio** [comment]
I have a similar problem.
I have two http server filters that add some extra data to mdc like logged user, user agent, language and so on... the controller are executed on another thread pool using @ ExecuteOn(IO) because of blocking database access.
I have customized the thread factory for that pool to copy the mdc entries from parent to child threads but seems that, sometimes it works and other not, randomly (or, to be more specific, some thread inherit and other not).

Here is my sample repo, https://github.com/francesco-monorchio/micronaut-samples/tree/inheritable-mdc

I am looking for a fully working solution. Thanks

**Comment by johncurrier** [comment]
Agreed, trying to find some documentation on how to glue MDC details into child threads started via @Async (or other techniques) is rather frustrating.

MdcInstrumenter seems to be the approach mentioned most often in issues here, but that's flagged as @Internal. There also doesn't seem to be any documentation about how that would be activated. ExecutorServiceInstrumenter seems to be the glue, but it too is @Internal. All of this seems to rely on on InvocationInstrumenterFactory, which is @Experimental.

@francesco-monorchio 's approach initially looked promising, but from what I can tell the MDC of whatever thread caused the thread pool size to increase gets used by subsequent threads.

Any chance of getting something documented?


Edit: I finally discovered that adding the following dependency in _build.gradle_ results in AWS Context variables being automatically populated in MDC and those are now being propagated to child threads (methods annotated with @Async). Logging that occurs in those child threads now includes those MDS values. I didn't have to do anything other than add this dependency:

`implementation "io.micronaut:micronaut-tracing"`

**Comment by crdhilep** [comment]
ServerRequestContext should able to share the context b/w threads.

Sample code snippet to get server context (getting trace ID header received in Request at web layer)

`
fun traceId(): String {
  var traceId: String? = UUID.randomUUID().toString()
    var currentRequest = ServerRequestContext.currentRequest<Any>()
      currentRequest.ifPresent {
      traceId = it.headers.get("traceId")
    } 
  return traceId.toString()
}`

**Comment by THARA40** [comment]
I have similar issue on Micronaut version 4.

`implementation "io.micronaut:micronaut-tracing"`

Above dependency not supported for Micronaut version 4.

I am looking solution for Micronaut version 4. Thanks

**Comment by ojebari-ma** [comment]
Can you please confirm that this is still an issue in the latest version of Micronaut?

**Comment by xpdota** [comment]
This seems to be fixed. I am able to get working MDC using the documentation at https://docs.micronaut.io/latest/guide/#httpFilterContextPropagation
